### DISCUSSION TITLE
emoji autocomplete `:smile:`

### DISCUSSION THREAD

**Discussion by avelino** [raphamorim/rio#114] [Ideas]
When you start typing `:`, do autocomplete for emojis

Does this feature make sense for Rio?

**Comment by raphamorim** [comment]
Does any other terminal implements it? (asking for reference as well)

**Reply by avelino** [reply]
I don't know, I shared what I wish the river had

**Comment by raphamorim** [comment]
Also, macos (as other platforms) allows use the emoji keyboard, wouldn't that make more sense for native platforms?

![image](https://github.com/raphamorim/rio/assets/3630346/fd911206-ef6d-4976-8de4-64491f58b2e0)

**Comment by OlshaMB** [comment]
I think this is an OS issue

**Reply by OlshaMB** [reply]
There are many apps for that

**Reply by OlshaMB** [reply]
For example there is this app for mac https://matthewpalmer.net/rocket/, also i rarely use emojies in terminal

**Comment by SeleneCosmia** [comment]
Currently Wezterm implements a [pop-over menu called CharSelect](https://wezfurlong.org/wezterm/config/lua/keyassignment/CharSelect.html#charselect) that includes a fuzzy search mode for unicode characters and emojis that are natively available within Wezterm itself. 

This is what it looks like without configuring the [char_select_fg_color](https://wezfurlong.org/wezterm/config/lua/config/char_select_fg_color.html#char_select_fg_color-rgba075-075-075-10) or [char_select_bg_color](https://wezfurlong.org/wezterm/config/lua/config/char_select_bg_color.html#char_select_bg_color-333333) options

![image](https://github.com/raphamorim/rio/assets/75750259/95e6a912-9ca8-4e59-bd82-c4e6d90cbeac)


I don't necessarily _*love*_ the current implementation in Wezterm, as it runs pretty slow and lags a lot if you scroll through the options too fast. I also wish the UI were more configurable with options like

- more configurable font rather than just a prebuilt font with a font size/ font color option
- options for setting a border for the pop-up menu *[relevant issue](https://github.com/wez/wezterm/issues/4239#issue-1878907749)* in wezterm's repo requesting exactly this

---
Here is the relevant [rust file](https://github.com/wez/wezterm/blob/914f18b19cc5bab3dd04f699d9cea17bbcf74d5b/wezterm-gui/src/termwindow/charselect.rs) that implements the CharSelect pop-up menu for reference 😛

**Reply by raphamorim** [reply]
It's very interesting, thank you for sharing it @SeleneCosmia . Took notes on it for when I start to work on #72
